packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -95,6 +98,45 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  // compare data between currentSyncData and previousSyncData and update it
+  previousSyncData.entries.filter(entry =>
+    _.findIndex(currentSyncData.deletedEntries, o => o.sys.id === entry.sys.id)
```

Comment:
What exactly are you trying to achieve with this logic? If I understand it correctly - you want to filter our the entries (and assets) which were deleted. In that case you would want to modify this logic like following:
```
_.findIndex(currentSyncData.deletedEntries, o => o.sys.id === entry.sys.id) !== -1
```